### PR TITLE
Added entities and cmakelists fix

### PR DESCRIPTION
**Pull request by makian123**
commit b4cf8ef9120611585408136b0b2b5a005c182b10
Author: makian <hohosmarko@gmail.com>
Date:   Mon Aug 5 17:16:57 2024 +0200

    Added time and crewmate class

    Changed the world update to take in a sf::Time copy instead of const ref

commit 8d2f28a9f092071d79f9eb2306fa21c2acb9a6e7
Author: makian <hohosmarko@gmail.com>
Date:   Mon Aug 5 17:15:46 2024 +0200

    Removed the header in add_executable function

### CODE REVIEW

File: src/crewmate.cpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+void Crewmate::Update(sf::Time deltaTime)
+{
+	// Quick check if adult or not
+	if (!adult && (world.GetTime().Year() - birthTimestamp.Year()) >= AdultYear)
+	{
+		adult = true;
+	}
+}
+
+Time Crewmate::Age() const { return world.GetTime() - birthTimestamp; }
```
Comment (lapinozz) [inline src/crewmate.cpp:20]: function should have a verb, like getAge()

File: src/crewmate.cpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#include "crewmate.hpp"
+#include "world.hpp"
+
+#include <iostream>
+
+Crewmate::Crewmate(const World &w, std::string_view name_, Gender gend)
```
Comment (lapinozz) [inline src/crewmate.cpp:6]: some of your parameters have a postfix and some don't

File: src/crewmate.cpp (hunk 1 above)
Comment (makian123) [reply src/crewmate.cpp:20]: I put it, but changed my mind, alr

File: src/crewmate.cpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+	: name(name_), gender(gend), world(w)
+{
+	birthTimestamp = world.GetTime();
+}
+void Crewmate::Update(sf::Time deltaTime)
+{
+	// Quick check if adult or not
+	if (!adult && (world.GetTime().Year() - birthTimestamp.Year()) >= AdultYear)
```
Comment (lapinozz) [inline src/crewmate.cpp:14]: I think the "isAdult()" function should compute that instead of saving it in a bool

File: src/ship.hpp [hunk 4]
```diff
@@ -40,9 +42,25 @@ class Ship : public Entity
 		grid.SetBlockType(9 * 30 + 19, {2, 1});
 	}
 
+	void CreateCrewmate(Crewmate &&crewmate) { 
+		crewmates.push_back(crewmate);
```
Comment (lapinozz) [inline src/ship.hpp:46]: you're copying the crewmate

**Comment by makian123** [comment]
Do not merge, issues with copy ctor

**Comment by makian123** [comment]
should be good now

**Review by lapinozz — APPROVED** [review]
looks good (enough)

**Pull request merged**